### PR TITLE
Pub 721 use api to fetch data.

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

https://tools.hmcts.net/jira/browse/PUB-721

Image 02-11-2021 at 09 33

**Does this PR introduce a breaking change?** (check one with "x")

[ ] Yes
[x] No

### CODE REVIEW

**Review by commitBlob — APPROVED** [review]

**Review by commitBlob — CHANGES_REQUESTED** [review]
Please also update mock data in a11y (see #140 )

**Review by johnoneill46 — DISMISSED** [review]

File: src/test/unit/requests/statusDescriptionRequests.test.ts [hunk 1]
```diff
@@ -4,60 +4,58 @@ import {StatusDescriptionRequests} from '../../../main/resources/requests/status
 import fs from 'fs';
 import path from 'path';
 
-const statusDescriptionRequests = new StatusDescriptionRequests();
+const searchDescriptionRequests = new StatusDescriptionRequests();
```
Comment (kurt-uk) [inline src/test/unit/requests/statusDescriptionRequests.test.ts:7]: it was right before

Comment (kurt-uk) [reply src/test/unit/requests/statusDescriptionRequests.test.ts:7]: the name of this file needs changing too

Comment (kurt-uk) [reply src/test/unit/requests/statusDescriptionRequests.test.ts:7]: in fact statusDescription is wrong too, this is supposed to be the event glossary right? so where did statusDescription come from?

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/test/unit/requests/statusDescriptionRequests.test.ts (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/test/unit/requests/statusDescriptionRequests.test.ts:7]: I don't remember where it came from. Do you want me to change all the relavent files to EventGlossary or CaseEventGlossary or CourtEventGlossary or CaseStatusDescription or something else? Reference link http://xhibit.justice.gov.uk/statusdescription.html

Comment (kurt-uk) [reply src/test/unit/requests/statusDescriptionRequests.test.ts:7]: i think either CaseEventGlossary or EventGlossary will be fine

File: src/test/unit/service/statusDescriptionService.test.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ const statusDescriptionService = new StatusDescriptionService();
 
 const statusDescriptionRequests = StatusDescriptionRequests.prototype;
 
+
```
Comment (kurt-uk) [inline src/test/unit/service/statusDescriptionService.test.ts:12]: remove pls

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/test/unit/requests/statusDescriptionRequests.test.ts (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/test/unit/requests/statusDescriptionRequests.test.ts]: everything change to CaseEventGlossary.

**Review by commitBlob — APPROVED** [review]

File: src/test/unit/controllers/caseEventGlossaryController.test.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+const stub = sinon.stub(CaseEventGlossaryService.prototype, 'generateCourtEventGlossaryObject').returns(caseEventGlossaryData);
+
+describe('Status Description Controller', () => {
+  it('should render the status description page', () =>  {
+    const i18n = {
+      'case-event-glossary': {},
+    };
+
+    stub.withArgs(1).returns(caseEventGlossaryData);
```
Comment (kurt-uk) [inline src/test/unit/controllers/caseEventGlossaryController.test.ts:22]: this method youre mocking doesnt take in any arguments so how come youre passing 1 in?

File: src/test/unit/controllers/caseEventGlossaryController.test.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    const response = { render: function() {return '';}} as unknown as Response;
+    const request = mockRequest(i18n);
+    request.query = {courtId: '1'};
+
+    const responseMock = sinon.mock(response);
+
+    responseMock.expects('render').once().withArgs('case-event-glossary');
```
Comment (kurt-uk) [inline src/test/unit/controllers/caseEventGlossaryController.test.ts:30]: this test is wrong and should be failing as you are not calling with those args

File: src/test/unit/requests/caseEventGlossaryRequests.test.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  });
+
+  it('Description fof First glossary must not be empty', () => {
+    stub.withArgs('/glossary').resolves({data: caseEventGlossaryData});
+    return searchDescriptionRequests.getCourtEventGlossaryList().then(data => {
+      expect(data[0].description).not.toBeNull();
+    });
+  });
+
+  let i = 0;
```
Comment (kurt-uk) [inline src/test/unit/requests/caseEventGlossaryRequests.test.ts:46]: can you move this into the test, it doesnt need to be declared outside it

File: src/test/unit/requests/caseEventGlossaryRequests.test.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      expect(data.length).toBe(49);
+    });
+  });
+
+  it('should return null if request fails', async () => {
+    stub.withArgs('/glossary').resolves(Promise.reject({response:{data: 'test error'}}));
+    expect(await searchDescriptionRequests.getCourtEventGlossaryList()).toHaveLength(0);
+  });
+
+  it('First glossary should be Adjourned', () => {
```
Comment (kurt-uk) [inline src/test/unit/requests/caseEventGlossaryRequests.test.ts:32]: try name your tests to start with "should" as thats the reason were using the alias `it` so they should read: `it('should do this or that...`

Comment (kurt-uk) [reply src/test/unit/requests/caseEventGlossaryRequests.test.ts:32]: same for all the rest of the tests

File: src/test/unit/requests/caseEventGlossaryRequests.test.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  it('All Glossary items must have name and description', () => {
+    stub.withArgs('/glossary').resolves({data: caseEventGlossaryData});
+    return searchDescriptionRequests.getCourtEventGlossaryList().then(data => {
+      expect(data[i].name).not.toBeNull();
+      expect(data[i].description).not.toBeNull();
+      i++;
+    });
+  });
+
+  it('should return null list of court event status', async () => {
```
Comment (kurt-uk) [inline src/test/unit/requests/caseEventGlossaryRequests.test.ts:56]: how come youve switched to async here? youve used sync pattern in the rest of the tests, we should try keep the test suite using the same pattern if poss

File: src/main/service/caseEventGlossaryService.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import {CaseEventGlossaryRequests} from '../resources/requests/caseEventGlossaryRequests';
+import {CourtService} from '../service/courtService';
+
+const caseEventGlossaryRequests = new CaseEventGlossaryRequests();
+
+export class CaseEventGlossaryService {
+
+  public async generateCourtEventGlossaryObject(): Promise<object> {
```
Comment (kurt-uk) [inline src/main/service/caseEventGlossaryService.ts:8]:
```suggestion
  public async generateCaseEventGlossaryObject(): Promise<object> {
```

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/test/unit/controllers/caseEventGlossaryController.test.ts (hunk 3 above)
Comment (junaidiqbalmoj) [reply src/test/unit/controllers/caseEventGlossaryController.test.ts]: removed.

File: src/test/unit/controllers/caseEventGlossaryController.test.ts (hunk 4 above)
Comment (junaidiqbalmoj) [reply src/test/unit/controllers/caseEventGlossaryController.test.ts]: this test is fixed now.

File: src/test/unit/requests/caseEventGlossaryRequests.test.ts (hunk 5 above)
Comment (junaidiqbalmoj) [reply src/test/unit/requests/caseEventGlossaryRequests.test.ts]: moved inside test.

File: src/test/unit/requests/caseEventGlossaryRequests.test.ts (hunk 7 above)
Comment (junaidiqbalmoj) [reply src/test/unit/requests/caseEventGlossaryRequests.test.ts:56]: text has been changed.

File: src/main/service/caseEventGlossaryService.ts (hunk 8 above)
Comment (junaidiqbalmoj) [reply src/main/service/caseEventGlossaryService.ts]: changed.

**Review by commitBlob — APPROVED** [review]

File: src/main/routes/routes.ts [hunk 9]
```diff
@@ -62,9 +62,11 @@ export default function(app: Application): void {
     regenerateSession);
   app.get('/login', passport.authenticate(authType, { failureRedirect: '/'}),
     regenerateSession);
-  app.get('/subscription-add', ensureAuthenticated, app.locals.container.cradle.subscriptionAddController.get);
-  app.post('/subscription-add', ensureAuthenticated, app.locals.container.cradle.subscriptionAddController.post);
-  app.get('/status-description', app.locals.container.cradle.statusDescriptionController.get);
+
+  app.get('/subscription-add', app.locals.container.cradle.subscriptionAddController.get);
+  app.post('/subscription-add', app.locals.container.cradle.subscriptionAddController.post);
```
Comment (johnoneill46) [inline src/main/routes/routes.ts:67]: Should 'ensureAuthenticated' not still be required for these screens?

Comment (junaidiqbalmoj) [reply src/main/routes/routes.ts]: added again. I mistakenly removed while fixing the code conflict.

**Review by junaidiqbalmoj — COMMENTED** [review]
done

**Review by johnoneill46 — APPROVED** [review]

**Review by commitBlob — DISMISSED** [review]

**Comment by commitBlob** [comment]
Approved.
Just a note that your coverage is 94% and remaining 6% is related to caseEventGlossaryRequests where you covered only 1 error state (while there are 3 different checks). 
It is really easy (copy paste) to cover those 6% from other request tests

**Review by kurt-uk — CHANGES_REQUESTED** [review]

**Review by commitBlob — APPROVED** [review]

**Review by johnoneill46 — APPROVED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Review by johnoneill46 — APPROVED** [review]

**Pull request merged**